Makefile

```diff
@@ -646,15 +649,16 @@ $(OBJ_TOOLS_SRC_DIR)/pson/Configfile: $(shell find src/tools/pson -type f)
 	rsync -a --delete src/tools/pson/ $(dir $@)
 	touch $@
 
-# PFPMP is my collection of FIRRTL passes.
-$(CMD_FIRRTL_GENERATE_TOP) $(CMD_FIRRTL_GENERATE_HARNESS): $(OBJ_TOOLS_BIN_DIR)/pfpmp/stamp $(CMD_SBT)
-	touch $@
+# barstools contains firrtl utilities useful for tapeout, like GenerateTop and
+# GenerateHarness.
+$(OBJ_TOOLS_BIN_DIR)/barstools/bin/%: src/tools/generate-sbt-wrapper $(OBJ_TOOLS_BIN_DIR)/barstools/stamp $(CMD_SBT)
+	@mkdir -p $(dir $@)
+	$< --sbt $(abspath $(CMD_SBT)) --project tapeout --sbtargs "-DFIRRTL_HOME=$(abspath $(FIRRTL_HOME))" --basedir $(abspath $(OBJ_TOOLS_BIN_DIR)/barstools) --main "barstools.tapeout.transforms.$(notdir $@)" --output $@
 
-$(OBJ_TOOLS_BIN_DIR)/pfpmp/stamp: $(shell find src/tools/pfpmp/src -type f) $(shell find $(CORE_DIR)/firrtl/src -type f)
+$(OBJ_TOOLS_BIN_DIR)/barstools/stamp: $(shell find src/tools/barstools/tapeout -type f) $(shell find $(FIRRTL_HOME)/src -type f)
 	rm -rf $(dir $@)
 	mkdir -p $(dir $@)
-	rsync -a --delete src/tools/pfpmp/ $(dir $@)
-	$(SCHEDULER_CMD) --make -- $(MAKE) FIRRTL_HOME=$(abspath $(CORE_DIR)/firrtl) CMD_SBT=$(abspath $(CMD_SBT)) -C $(dir $@)
+	rsync -a --delete src/tools/barstools/ $(dir $@)
```

Comment:
Yes, I agree. It might be nice too to have a mode where we don't touch the main directories and try to do as much work in object folders as possible (e.g. running multiple PLSI instances simultaneously from one folder), so I added a mode that lets us do that if the source dir is read-only.